activerecord/lib/active_record/connection_adapters/abstract/quoting.rb

```diff
@@ -17,7 +22,11 @@ def quote(value)
       # SQLite does not understand dates, so this method will convert a Date
       # to a String.
       def type_cast(value, column = nil)
+        value = id_value_for_database(value) if value.is_a?(Base)
+
         if value.respond_to?(:quoted_id) && value.respond_to?(:id)
+          ActiveSupport::Deprecation.warn \
```

Comment:
I understand the problem but in this case here `quoted_id` was never called and the deprecation message say it was. If using `value.id` is a bug we should just fix the bug and remove the deprecation.